### ISSUE TITLE
Item template is no longer visible after adding Asset elements

### ISSUE THREAD

**Issue by PawcioCNCProfi** [NuGet/Home#8957]
- After adding the `Asset` elements for nuget packages. The template is no longer listed in list of available item to add to projects.

- I also don't understand why do I need to set `repositoryId` in template properties to VSIX project id, essentially creating circular dependency between those project. 

- The "2. Place your `nupkg` files in a folder called `Packages` within the VSIX." instruction is also not clear. Is putting the folder in file system enough? Do I need to also include it in the project definition? Should I copy it to output directory or set the files as resources? 
- Where exactly do I put the `&lt;Reference ...&gt;` element mentioned in "Best practices"?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: bbf9be6a-e7b0-b9b6-a3e8-8b5de3c6900d
* Version Independent ID: c12505c9-6f14-7111-2578-5c2664b4515d
* Content: [NuGet Packages in Visual Studio templates](https://docs.microsoft.com/en-us/nuget/visual-studio-extensibility/visual-studio-templates)
* Content Source: [docs/visual-studio-extensibility/Visual-Studio-Templates.md](https://github.com/NuGet/docs.microsoft.com-nuget/blob/master/docs/visual-studio-extensibility/Visual-Studio-Templates.md)
* Product: **nuget**
* GitHub Login: @karann-msft
* Microsoft Alias: **karann**

**Comment by PawcioCNCProfi** [comment]
I did not managed to accomplish installation of packages as described [in the article](https://docs.microsoft.com/en-us/nuget/visual-studio-extensibility/visual-studio-templates). But I found other way to do this.

I've created [custom template wizard](https://docs.microsoft.com/en-us/visualstudio/extensibility/how-to-use-wizards-with-project-templates?view=vs-2019). 

Then I placed this method in `IWizzard` implementation:

```c#
public bool InstallNuGetPackage(Project project, string package, string version) {
    bool installedPkg = true;
    try {
        var componentModel = (IComponentModel)Package.GetGlobalService(typeof(SComponentModel));
        var installerServices = componentModel.GetService<IVsPackageInstallerServices>();
        if (!installerServices.IsPackageInstalled(project, package)) {
            var installer = componentModel.GetService<IVsPackageInstaller>();
            installer.InstallPackage(null, project, package, version, false);
        }
    } catch (Exception ex) {
        installedPkg = false;
    }
    return installedPkg;
}
```

This requires installing additional packages in the wizard project. I've installed following ones:

- **NuGet.Packaging.Core** - not sure if this one is actually needed
- **NuGet.VisualStudio**

Here are namespaces used with the `IWizzard` implementation.

```c#
using EnvDTE;
using Microsoft.VisualStudio.ComponentModelHost;
using Microsoft.VisualStudio.TemplateWizard;
using System;
using System.Collections.Generic;
using System.Windows;
using NuGet.VisualStudio;
using Microsoft.VisualStudio.Shell;
```

Above method is based on flowing topics:

- https://stackoverflow.com/questions/49303139/how-to-install-nuget-packages-using-envdte-dte
- https://stackoverflow.com/questions/41225508/how-to-use-ivspackagesourceprovider-provider

Finally I use the method in **PorjectItemFinishedGenerating()** method of the `IWizzard`.

```c#
public void ProjectItemFinishedGenerating(ProjectItem projectItem) {
    ThreadHelper.ThrowIfNotOnUIThread();
    var i = projectItem;
    var p = i.ContainingProject;
    InstallNuGetPackage(p, "CNCProfi.Logging", "1.0.7");
    InstallNuGetPackage(p, "CNCProfi.BaseUtils", "1.0.21");
    InstallNuGetPackage(p, "CNCProfi.General", "1.0.12.1");
    InstallNuGetPackage(p, "CNCProfi.Geometry", "1.0.1");
    InstallNuGetPackage(p, "CNCProfi.GraphicsWPF", "1.0.1.1");
    InstallNuGetPackage(p, "CNCProfi.ProfiApp", "1.0.3");
    InstallNuGetPackage(p, "CNCProfi.CADDocument", "1.0.0.1");
    InstallNuGetPackage(p, "CNCProfi.Vectors", "1.0.1");
}
```

This method installs packages from configured sources and does not require to include NuGet packages in the extension project. It can be also configured differently and does not require for a template project to be dependent on VISX extension where it is used. So generally I found it superior to the method presented in the article.

**Comment by nkolev92** [comment]
Hey @PawcioCNCProfi 

Are you still blocked on any of these questions?

**Comment by kartheekp-ms** [comment]
@PawcioCNCProfi - Are you still blocked on any of these questions?

@nkolev92 - I think we can improve NuGet documentation based on [above comment](https://github.com/NuGet/Home/issues/8957#issuecomment-567020111).

**Comment by PawcioCNCProfi** [comment]
> Hey @PawcioCNCProfi
> 
> Are you still blocked on any of these questions?

I posted previous reply as an alternative solution. So original questions are no longer relevant for me.

**Comment by nkolev92** [comment]
Thanks for following up. 

Closing.
